int64 frequencyIn
---
int64 frequencyOut